### PR TITLE
Pass WS URL to connect to override default

### PR DESCRIPTION
**Pull request by hjheath**

### CODE REVIEW

**Comment by rfreitas** [comment]
probably good to have a safe get: http://adripofjavascript.com/blog/drips/making-deep-property-access-safe-in-javascript.html

this one looks tiny: https://github.com/wakecoder/sget

**Comment by hjheath** [comment]
The safe get does not work with the Webpack define plugin because define replaces variables inline rather than creating an object. All the safe get libraries rely on the object actually existing. The solution is to use a top level name i.e. `ONFIDO_ENV` rather than `process.env.ONFIDO_ENV`.

**Comment by rfreitas** [comment]
Better then, less code to add.

File: src/connect/socket.js [hunk 1]
```diff
@@ -8,9 +8,9 @@ const { setWebSocketError, setAuthenticated } = actions
 
 export default class Socket {
 
-  connect(jwt) {
+  connect(socket_url=constants.SOCKET_URL, jwt) {
```
Comment (rfreitas) [inline src/connect/socket.js]: optional parameters are better placed at the end of the argument list

**Review by rfreitas — COMMENTED** [review]
looks good, apart from minor points

File: src/connect/index.js [hunk 2]
```diff
@@ -12,12 +14,18 @@ function setSupport() {
   setGumSupport(supportsGetUserMedia)
 }
 
-export default function connect(url, jwt) {
+function constructUrl(jwt, socket_url) {
```
Comment (rfreitas) [inline src/connect/index.js]: camelcase `socket_url`
this could also be written this way: `const constructUrl = (jwt, socketUrl) => {...}`

**Review by rfreitas — APPROVED** [review]
looks good

**Pull request merged**